shims/spark310/src/main/scala/org/apache/spark/sql/rapids/shims/spark310/RapidsShuffleInternalManager.scala

```diff
@@ -36,6 +36,7 @@ class RapidsShuffleInternalManager(conf: SparkConf, isDriver: Boolean)
       endPartition: Int,
       context: TaskContext,
       metrics: ShuffleReadMetricsReporter): ShuffleReader[K, C] = {
-    getReaderInternal(handle, 0, Int.MaxValue, startPartition, endPartition, context, metrics)
+    getReaderInternal(handle, startMapIndex, endMapIndex, startPartition, endPartition, context,
```

Comment:
This change makes sense to me. I'd assume without it we get extra blocks and the results are incorrect, when the reader wants to be one of the custom shuffle readers.